File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(n)
    response = ''
    response << 'Pling' if divisible? n, 3
    response << 'Plang' if divisible? n, 5
    response << 'Plong' if divisible? n, 7

    return n.to_s if response.length == 0

    response
  end

  def self.divisible?(m, n)
    m % n == 0
  end
end

File: javascript/raindrops/raindrops.js
'use strict';

function Raindrops() {}

Raindrops.prototype.convert = function (num) {
  let text = "";

  if (num % 3 === 0) { text += 'Pling' }
  if (num % 5 === 0) { text += 'Plang' }
  if (num % 7 === 0) { text += 'Plong' }
  if (text.length > 0) { return text; }

  return num.toString();
}

module.exports = Raindrops;
